File: backend/Entities/Playlist.cs
using System.ComponentModel.DataAnnotations;
using MassTransit;
using SongAppApi.Helpers.Enumerators;

namespace SongAppApi.Entities
{
    public class Playlist
    {
        [Key]
        public Guid Id { get; set; } = NewId.NextSequentialGuid();
        public string Name { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public Guid CreatedById { get; set; }
        public Account CreatedBy { get; set; }
        public PlaylistVisibility Visibility { get; set; } = PlaylistVisibility.Private;
        public File? Image { get; set; }
        public List<Song> Songs { get; set; } = new List<Song>();
        public List<Account> SavedByAccounts { get; set; } = new List<Account>();
    }
}

File: frontend/src/models/playlist.ts
import type { Song } from "./song";

export const PlaylistVisibility = {
    Private: "Private",
    Unlisted: "Unlisted",
    Public: "Public",
} as const;
export type PlaylistVisibility =
    typeof PlaylistVisibility[keyof typeof PlaylistVisibility];

export function visibilityLabel(v: PlaylistVisibility): string {
    switch (v) {
        case PlaylistVisibility.Private: return "Private";
        case PlaylistVisibility.Unlisted: return "Unlisted";
        case PlaylistVisibility.Public: return "Public";
        default: return "Unknown";
    }
}

export interface Playlist {
    id: string,
    name: string,
    createdAt: string,
    updatedAt: string,
    visibility: PlaylistVisibility,
    songs: Song[],
    createdBy?: {
        id: string,
        userName?: string,
    },
}

export interface CreatePlaylistRequest {
    name: string,
    songIds: string[],
    visibility?: PlaylistVisibility,
}

export interface UpdatePlaylistRequest {
    id: string,
    name: string,
    songIds: string[],
}

export interface DeletePlaylistRequest {
    id: string,
}